seed/php-sdk/imdb/clientName/src/Core/Client/RetryMiddleware.php

```diff
@@ -0,0 +1,192 @@
+<?php
+
+namespace Seed\Core\Client;
+
+use GuzzleHttp\Exception\ConnectException;
+use GuzzleHttp\Promise as P;
+use GuzzleHttp\Promise\PromiseInterface;
+use Psr\Http\Message\RequestInterface;
+use Psr\Http\Message\ResponseInterface;
+use Throwable;
+
+class RetryMiddleware
+{
+    private const RETRY_STATUS_CODES = [408, 429];
+
+    /**
+     * @var callable(RequestInterface, array): PromiseInterface
+     * @phpstan-ignore missingType.iterableValue
+     */
+    private $nextHandler;
+
+    /**
+     * @var array{
+     *     maxRetries: int,
+     *     baseDelay: int,
+     * }
+     */
+    private array $options;
+
+    /**
+     * @param callable $nextHandler
+     * @param array{
+     *     maxRetries?: int,
+     *  }|null $options
```

Comment:
nit: We can use an optional parameter instead of `|null` here.
```suggestion
     * @param array{
     *     maxRetries?: int,
     *  } ?$options
```

Reply:
Done